src/core/net/dns_headers.hpp

```diff
@@ -1178,6 +1232,37 @@ class TxtRecord : public ResourceRecord
                         uint8_t *      aTxtBuffer,
                         uint16_t &     aTxtBufferSize) const;
 
+    /**
```

Comment:
How about we define a typedef for `TxtIterator` here?
```
typedef otDnsTxtIterator TxtIterator;
```
I like (and persoanlly always try) to re-define `ot{Type}` if/when we can and not use them in core module. 😃

Reply:
SGTM, updated.